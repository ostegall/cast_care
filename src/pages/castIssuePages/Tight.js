import { Text, View, Pressable } from 'react-native';
import styles from '../../styles';

export default function Tight({navigation}) { //Solutions for user that has a tight cast
    return (
        <View style={{flex: 1, paddingTop: 160}}>
          <Text style={styles.header_text }>
            Problem Description:
          </Text>
          <Text style={styles.page_text}>
            There should be room enough for 2 fingers to slide between patient's cast and patient's skin. If there is not enough and the cast is too tight or if there is not enough room between the cast/splint and patient's skin, it could restrict motion.
          </Text>
          <Text style={styles.header_text }>
            Why this is a Problem:
          </Text>
          <Text style={styles.page_text}>
            If the cast is too tight, it could cut off circulation and cause other issues. It is important it is not too tight, elevate extremity 6-8 hours and anytime it is throbbing.
          </Text>
          <Text style={styles.header_text }>
            Correcting the Problem:
          </Text>
          <Text style={styles.page_text}>
            Make sure fingers/toes are not white, swollen, shiny or do not blanch after elevation. If they do, schedule an appointment.
          </Text>
          <Text style={styles.header_text }>
            Who to Contact:
          </Text>
          <Text style={styles.page_text}>
            Follow-up with the office by calling 984-974-5700 to schedule a same-day appointment or come into the Orthonow walk-in clinic. If patient is not in the NC Triangle area, please go to your local ED.
          </Text>
          <View style={{alignItems: 'center', padding: 50}}>
            <Pressable style={styles.button}
              onPress= {() => {
                navigation.navigate("Info") //Navigates to page for finding nearest hospital
              }}
            >
              <Text style={styles.text}>
                Find your nearest location
              </Text>
            </Pressable>
          </View>
        </View>
      )
}