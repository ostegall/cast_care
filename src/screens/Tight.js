import React from 'react';
import { Text, View } from 'react-native';

class Tight extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <View>
        <Text>Too Tight? Selection
        </Text>
      </View>
    )
  }
}

export default Tight