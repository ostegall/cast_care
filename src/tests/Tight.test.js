import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'

import Tight from '../pages/castIssuePages/Tight';

//Makes sure app component renders correctly without crashing
it("renders without crashing", () => {
  shallow(<Tight />);
});

//Snapshot test to ensure UI stays consistent
it('renders correctly', () => {
  const tree = renderer.create(<Tight />).toJSON();
  expect(tree).toMatchSnapshot();
});