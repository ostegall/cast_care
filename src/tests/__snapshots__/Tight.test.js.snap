// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 23,
        "fontWeight": "bold",
        "letterSpacing": 0.5,
        "textAlign": "center",
      }
    }
  >
    Problem Description:
  </Text>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 17,
        "letterSpacing": 0.5,
        "marginBottom": 5,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 5,
      }
    }
  >
    There should be room enough for 2 fingers to slide between patient's cast and patient's skin. If there is not enough and the cast is too tight or if there is not enough room between the cast/splint and patient's skin, it could restrict motion.
  </Text>
  <Text
    style={
      Object {
        "fontSize": 23,
        "fontWeight": "bold",
        "letterSpacing": 0.5,
        "textAlign": "center",
      }
    }
  >
    Why this is a Problem:
  </Text>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 17,
        "letterSpacing": 0.5,
        "marginBottom": 5,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 5,
      }
    }
  >
    If the cast is too tight, it could cut off circulation and cause other issues. It is important it is not too tight, elevate extremity 6-8 hours and anytime it is throbbing.
  </Text>
  <Text
    style={
      Object {
        "fontSize": 23,
        "fontWeight": "bold",
        "letterSpacing": 0.5,
        "textAlign": "center",
      }
    }
  >
    Correcting the Problem:
  </Text>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 17,
        "letterSpacing": 0.5,
        "marginBottom": 5,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 5,
      }
    }
  >
    Make sure fingers/toes are not white, swollen, shiny or do not blanch after elevation. If they do, schedule an appointment.
  </Text>
  <Text
    style={
      Object {
        "fontSize": 23,
        "fontWeight": "bold",
        "letterSpacing": 0.5,
        "textAlign": "center",
      }
    }
  >
    Who to Contact:
  </Text>
  <Text
    style={
      Object {
        "alignItems": "center",
        "fontSize": 17,
        "letterSpacing": 0.5,
        "marginBottom": 5,
        "marginLeft": 20,
        "marginRight": 20,
        "marginTop": 5,
      }
    }
  >
    Follow-up with the office by calling 984-974-5700 to schedule a same-day appointment or come into the Orthonow walk-in clinic. If patient is not in the NC Triangle area, please go to your local ED.
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "height": 83,
          "marginLeft": 24,
          "marginRight": 23,
          "marginTop": 20,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4B9CD3",
            "borderRadius": 10,
            "elevation": 2,
            "flexDirection": "row",
            "height": 80,
            "justifyContent": "center",
            "minWidth": 88,
            "paddingLeft": 16,
            "paddingRight": 16,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.35,
            "shadowRadius": 5,
            "width": 120,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 14,
              "letterSpacing": 0.25,
              "textAlign": "center",
            }
          }
        >
          Find your nearest location
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#4B9CD3",
            "borderRadius": 10,
            "elevation": 2,
            "flexDirection": "row",
            "height": 80,
            "justifyContent": "center",
            "minWidth": 88,
            "paddingLeft": 16,
            "paddingRight": 16,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.35,
            "shadowRadius": 5,
            "width": 120,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 14,
              "letterSpacing": 0.25,
              "textAlign": "center",
            }
          }
        >
          Show image of tight cast
        </Text>
      </View>
    </View>
  </View>
</View>
`;
